db/main.py

```diff
@@ -0,0 +1,77 @@
+import base64
+import sys
+from sqlalchemy import create_engine, Table, MetaData, Column, String
+from sqlalchemy.sql import select
+
+
+class MagpieData(object):
+    def __init__(self, url, metadata=None):
+        self.url = url
+        if(metadata is not None):
+            self.rowcount = 1
+            self.metadata = str(metadata._row[0])
+        else:
+            self.rowcount = 0
+            self.metadata = metadata
+
+
+class DbUtils(object):
+    def establish_db_connection(self):
+        # Establish connection
+        connectionString = 'postgres://' + base64.b64decode(self.username) + ':' + base64.b64decode(self.password) + '@' + base64.b64decode(self.hostName) + '/' + self.dbName
+        self.engine = create_engine(connectionString)
+        self.connection = self.engine.connect()
+
+    def __init__(self):
+        credentials = []
+        # Open credentials file. Should be stored as "username:password"
+        with open('credentials.txt') as f:
+            for line in f:
+                line = line.split("\n", 1)[0]
+                credentials.append(line.split(':', 1))
+
+        self.username = base64.b64encode(credentials[0][0])
+        self.password = base64.b64encode(credentials[0][1])
+        self.dbName = credentials[1][1]
+        self.tableName = credentials[2][1]
+        self.hostName = base64.b64encode(credentials[3][1])
+        try:
+            self.establish_db_connection()
+        except:
+            print "Oops! Could not establish connection to database at " + base64.b64decode(self.hostName)
+            print "Terminating program"
+            sys.exit(1)
```

Comment:
We should never be doing sys.exit. We don't want to stop people from using the library. Just throw an exception to the app, and the app can handle it